src/realm/set.hpp

```diff
@@ -1240,13 +1240,25 @@ inline size_t LnkSet::find_any(Mixed value) const
 {
     if (value.is_null())
         return not_found;
-    if (value.get_type() != type_Link)
-        return not_found;
-    size_t found = find(value.get<ObjKey>());
-    if (found != not_found) {
-        found = real2virtual(found);
+
+    auto find_obj_key = [this](ObjKey objkey) {
+        size_t found = find(objkey);
```

Comment:
This find method makes the translation to virtual index. Just use it directly below. This seems to have been an error before as well. Do we actually have a test involving this? I played around to verify my assumptions - may as well push my findings.